storage/grpc_reader_bidi.go

```diff
@@ -0,0 +1,531 @@
+// Copyright 2025 Google LLC
```

Comment:
Could you skip the move to a new file for this PR? It makes it hard to understand the diff.

Also I was thinking we could just centralize all download logic in grpc_reader.go.

Reply:
I'll move it back for this PR. 

I think we should keep more separation - MRD seems to be a unit of code that doesn't interact with the other read code too much.

Reply:
Ok I also removed the readIDGenerator piece, I will open a separate pr for that. I know it's a bunch of little changes together which makes it hard to review! Let me know if you want me to piece this out even more.